Linphone/Core/CallFlowControl.cs

```diff
@@ -0,0 +1,376 @@
+using Linphone.Views;
+using Serilog;
+using System;
+using System.Threading.Tasks;
+using System.Web;
+using Windows.Storage;
+
+namespace BelledonneCommunications.Linphone.Core
+{
+    // Todo: Mention: Sip login on shayan side, means that "I am ready for accept incoming calls" and does nothing to outgoing calls.
+    internal class CallFlowControl
+    {
+        private static readonly CallFlowControl _instance = new CallFlowControl();
+        public static CallFlowControl Instance
+        {
+            get
+            {
+                return _instance;
+            }
+        }
+
+        private CallFlowControl()
+        {
+            ApplicationDataContainer localSettings = ApplicationData.Current.LocalSettings;
+            object settingValue = localSettings.Values["PanelUrl"];
+            string panelBaseUrl = settingValue == null ? "http://10.19.82.133:9012" : settingValue as string;
+
+            _logger = Log.Logger.ForContext("SourceContext", nameof(CallFlowControl));
+            _coreClient = new CoreHttpClient(panelBaseUrl);
+
+            CallContext = new CallContext();
+            AgentProfile = new PhoneProfile(panelBaseUrl);
+        }
+
+        public async Task<CallsCommandServiceInitiateIncomingResponse> InitiateIncomingCallAsync(string callerPhoneNumber, string inboundService)
+        {
+            try
+            {
+                _logger.Information("Initiation incoming call from: {CallerNumber}.", callerPhoneNumber);
+
+                CallContext.CallState = CallState.Ringing;
+                CallContext.CallerNumber = callerPhoneNumber;
+                CallContext.InboundService = inboundService;
+                CallContext.CalleeNumber = AgentProfile.SipPhoneNumber;
+                CallContext.Direction = CallDirection.Incoming;
+                CallContext.CallId = default;
+
+                CallsCommandServiceInitiateIncomingResponse response =
+                    await _coreClient.InitiateIncomingCallAsync(callerPhoneNumber: callerPhoneNumber,
+                                                                agentPhoneNumber: AgentProfile.SipPhoneNumber,
+                                                                inboundService: inboundService);
+                if (response != null)
+                {
+                    CallContext.CallId = response.Data.Id;
+                }
+
+                _logger.Information("Incoming call initiation successfully done with call id: {CallId}.", CallContext.CallId.ToString());
+
+                return response;
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, "Internal error during call initation.");
+                return null;
+            }
+        }
+
+        // Todo: We should prevent to submit 2 call record for missed calls. Please check it.
+        public async Task<CallsCommandServiceInitiateOutgoingResponse> InitiateOutgoingCallAsync(string calleePhoneNumber, string inboundService)
```

Comment:
Please add API comment for this method

* What is meaning of inboundService?
* What is format of calleePhoneNumber?

Reply:
Requested comments added to the code. please read and verify if it was clear and straight-forward.